wlm/src/main/java/ai/djl/serving/wlm/util/SageMakerCompatibility.java

```diff
@@ -0,0 +1,123 @@
+            properties.setProperty("max_request_size", maxRequestSize);
+
+            if (maxPayloadMb != null) {
+                logger.warn(
+                        "Both SAGEMAKER_MAX_REQUEST_SIZE and SAGEMAKER_MAX_PAYLOAD_IN_MB are set."
+                                + " Using SAGEMAKER_MAX_REQUEST_SIZE={} and ignoring"
+                                + " SAGEMAKER_MAX_PAYLOAD_IN_MB={}",
+                        maxRequestSize,
+                        maxPayloadMb);
+            }
+        } else if (maxPayloadMb != null) {
+            try {
+                long payloadBytes = Long.parseLong(maxPayloadMb) * 1024 * 1024;
+                logger.info(
+                        "SageMaker compatibility - translating SAGEMAKER_MAX_PAYLOAD_IN_MB={} to"
+                                + " max_request_size={} bytes",
+                        maxPayloadMb,
+                        payloadBytes);
+                properties.setProperty("max_request_size", String.valueOf(payloadBytes));
+            } catch (NumberFormatException e) {
+                logger.warn("Invalid SAGEMAKER_MAX_PAYLOAD_IN_MB value: {}", maxPayloadMb);
+            }
+        }
+    }
+
+    /**
+     * Applies SageMaker compatibility for model-level configurations. Called during
+     * ModelInfo.loadServingProperties().
+     *
+     * @param properties the properties to modify
+     */
+    public static void applyModelCompatibility(Properties properties) {
+        String numWorkers = System.getenv("SAGEMAKER_NUM_MODEL_WORKERS");
+        if (numWorkers != null) {
+            logger.info(
+                    "SageMaker compatibility - translating SAGEMAKER_NUM_MODEL_WORKERS={} to"
+                            + " minWorkers/maxWorkers",
+                    numWorkers);
+            properties.setProperty("minWorkers", numWorkers);
+            properties.setProperty("maxWorkers", numWorkers);
+        }
+
+        String startupTimeout = System.getenv("SAGEMAKER_STARTUP_TIMEOUT");
+        if (startupTimeout != null) {
+            logger.info(
+                    "SageMaker compatibility - translating SAGEMAKER_STARTUP_TIMEOUT={} to"
+                            + " model_loading_timeout",
+                    startupTimeout);
+            properties.setProperty("option.model_loading_timeout", startupTimeout);
+        }
+
+        String predictTimeout = System.getenv("SAGEMAKER_MODEL_SERVER_TIMEOUT_SECONDS");
+        if (predictTimeout != null) {
+            logger.info(
+                    "SageMaker compatibility - translating"
+                            + " SAGEMAKER_MODEL_SERVER_TIMEOUT_SECONDS={} to predict_timeout",
+                    predictTimeout);
+            properties.setProperty("option.predict_timeout", predictTimeout);
+        }
+
+        String defaultAccept = System.getenv("SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT");
+        if (defaultAccept != null) {
+            String entryPoint = properties.getProperty("option.entryPoint");
```

Comment:
This makes you need to move `SageMakerCompatibility.applyModelCompatibility(prop)` in `ModelInfo.java` after line 915. Because if user set `OPTION_ENTRYPOINT` env, it is not populated to properties yet.